crates/biome_js_analyze/src/lint/nursery/use_readonly_class_properties.rs

```diff
@@ -0,0 +1,449 @@
+
+            let replace_parameter = make::ts_property_parameter(
+                parameter.decorators(),
+                replace_modifiers,
+                parameter.formal_parameter().expect("REASON"),
+            );
+
+            mutation.replace_node_discard_trivia(parameter.clone(), replace_parameter);
+        }
+
+        Some(JsRuleAction::new(
+            ctx.metadata().action_category(ctx.category(), ctx.group()),
+            ctx.metadata().applicability(),
+            markup! { "Add `readonly` decorator." }.to_owned(),
+            mutation,
+        ))
+    }
+}
+
+/// Rule's options
+#[derive(Clone, Debug, Default, Deserialize, Deserializable, Eq, PartialEq, Serialize)]
+#[cfg_attr(feature = "schema", derive(schemars::JsonSchema))]
+#[serde(rename_all = "camelCase", deny_unknown_fields, default)]
+pub struct ReadonlyClassPropertiesOptions {
+    /// Determines whether all class properties should be analyzed.
+    ///
+    /// - **`true`**: Checks all class properties, including `public`, `protected`, and `private`.
+    /// - **`false`**: Only `private` properties are checked. This is the default behavior.
+    #[serde(default, skip_serializing_if = "is_default")]
+    pub check_all_properties: bool,
+}
+
+fn is_default<T: Default + Eq>(value: &T) -> bool {
+    value == &T::default()
+}
+
+type TextRangeWithText = (TextRange, Text, SyntaxNode<JsLanguage>);
+
+fn filter_mutable_class_property(
+    member: &AnyJsClassMember,
+    private_only: bool,
+) -> Option<TextRangeWithText> {
+    let property_class_member = member.as_js_property_class_member()?;
+
+    if property_class_member
+        .modifiers()
+        .iter()
+        .any(|x| TsReadonlyModifier::cast(x.into_syntax()).is_some())
+    {
+        return None;
+    }
+
+    let name = property_class_member.name().ok()?;
+    let text_range_and_name = Some((
+        name.range(),
+        name.to_trimmed_text(),
+        property_class_member.syntax().clone(),
+    ));
+
+    if !private_only {
+        return text_range_and_name;
+    }
+
+    let is_es_private = matches!(
+        member.name().ok()??,
+        AnyJsClassMemberName::JsPrivateClassMemberName(_)
+    );
+    let is_ts_private = property_class_member.modifiers().iter().any(|x| {
+        TsAccessibilityModifier::cast(x.into_syntax()).is_some_and(|modifier| modifier.is_private())
+    });
+
+    if is_es_private || is_ts_private {
+        return text_range_and_name;
+    }
+    None
+}
+
+fn filter_mutable_constructor_parameters(
+    class_declaration: &JsClassDeclaration,
+    private_only: bool,
+) -> Option<Vec<TextRangeWithText>> {
+    let constructor = class_declaration
+        .members()
+        .iter()
+        .find_map(|member| match member {
+            AnyJsClassMember::JsConstructorClassMember(member) => Some(member),
+            _ => None,
+        })?;
+
+    let constructor_params = constructor.parameters().ok()?;
+    let params: Vec<TextRangeWithText> = constructor_params
```

Comment:
```suggestion
    let params: Vec<_> = constructor_params
```

`collect()` is good and most of the time doesn't require the explicit type. Add a type only when the compiler tells you to